vehicle/homeassistant.go

```diff
@@ -0,0 +1,320 @@
+		embed:        &cc.embed,
+		Helper:       request.NewHelper(log),
+		conf:         cc,
+		startScript:  cc.Services.Start,
+		stopScript:   cc.Services.Stop,
+		wakeupScript: cc.Services.Wakeup,
+	}, nil
+}
+
+// -----------------------------
+// Internal helper functions
+// -----------------------------
+
+// Calls /api/states/<entity> and returns .state
+func (v *HomeAssistant) getState(entity string) (string, error) {
+	uri := fmt.Sprintf("%s/api/states/%s", strings.TrimSuffix(v.conf.Host, "/"), entity)
+	req, err := request.New(http.MethodGet, uri, nil, map[string]string{
+		"Authorization": "Bearer " + v.conf.Token,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	var resp struct {
+		State string `json:"state"`
+	}
+	if err = v.DoJSON(req, &resp); err != nil {
+		return "", err
+	}
+	return resp.State, nil
+}
+
+// Calls script.<name> as a service call without payload
+func (v *HomeAssistant) callScript(script string) error {
+	if script == "" {
+		return api.ErrNotAvailable
+	}
+	domain, name, ok := strings.Cut(script, ".")
+	if !ok { // kein Punkt gefunden
+		return fmt.Errorf("homeassistant: invalid script name '%s'", script)
+	}
+	uri := fmt.Sprintf("%s/api/services/%s/%s", v.conf.Host, domain, name)
+
+	req, err := request.New(http.MethodPost, uri, bytes.NewBuffer([]byte("{}")), map[string]string{
+		"Authorization": "Bearer " + v.conf.Token,
+		"Content-Type":  "application/json",
+	})
+	if err != nil {
+		return err
+	}
+	_, err = v.DoBody(req)
+	return err
+}
+
+// -----------------------------
+// Implementation of API interfaces
+// -----------------------------
+
+// generic helpers for fetching and parsing sensor values
+func (v *HomeAssistant) getFloatSensor(entity string) (float64, error) {
+	if entity == "" {
+		return 0, api.ErrNotAvailable
+	}
+	s, err := v.getState(entity)
+	if err != nil {
+		return 0, err
+	}
+	return strconv.ParseFloat(s, 64)
+}
+
+func (v *HomeAssistant) getIntSensor(entity string) (int64, error) {
+	if entity == "" {
+		return 0, api.ErrNotAvailable
+	}
+	s, err := v.getState(entity)
+	if err != nil {
+		return 0, err
+	}
+	return strconv.ParseInt(s, 10, 64)
+}
+
+func (v *HomeAssistant) getBoolSensor(entity string) (bool, error) {
+	if entity == "" {
+		return false, api.ErrNotAvailable
+	}
+	s, err := v.getState(entity)
+	if err != nil {
+		return false, err
+	}
+	state := strings.ToLower(s)
+	return state == "on" || state == "true" || state == "1" || state == "active", nil
+}
+
+func (v *HomeAssistant) getTimeSensor(entity string) (time.Time, error) {
+	if entity == "" {
+		return time.Time{}, api.ErrNotAvailable
+	}
+	s, err := v.getState(entity)
+	if err != nil {
+		return time.Time{}, err
+	}
+	if ts, err := strconv.ParseInt(s, 10, 64); err == nil {
+		return time.Unix(ts, 0), nil
+	}
+	t, err := time.Parse(time.RFC3339, s)
+	return t, err
+}
+
+// Soc returns state of charge [%]
+func (v *HomeAssistant) Soc() (float64, error) {
+	return v.getFloatSensor(v.conf.Sensors.Soc)
```

Comment:
Dito..  all thes eoptional readings MUST be decorated

Reply:
decorated